lib/src/core/verifier/policy/opa/opa.go

```diff
@@ -0,0 +1,78 @@
+package main
+
+import "C"
+
+import (
+	"context"
+	"encoding/json"
+	"strings"
+
+	"github.com/open-policy-agent/opa/rego"
+	"github.com/open-policy-agent/opa/storage/inmem"
+)
+
+//export evaluateGo
+func evaluateGo(policy string, data string, input string) *C.char {
+	// Deserialize the message in json format
+	input_map := make(map[string]interface{})
+	err := json.Unmarshal([]byte(input), &input_map)
+	if err != nil {
+		return C.CString("Error:: " + err.Error())
```

Comment:
this function returns two types of CStrings: if it begins with `Error:: ` then it's an error, otherwise it's a "good" result. Instead, I suggest the following:

1. Extract an internal "classic" Go function that returns `string, err` (where err is nil if no error, as usual). No CString involved.
2. The exported `evaluteGo` function will call the internal one, and then format the single C-compatible return value, including the unescaping of `\\u003e` etc.
3. I suggest the return CString to start with `Error:: ....` or `Ok:: ...` so it's more explicit.